src/diff.c

```diff
@@ -296,6 +292,8 @@ diff_common_read(struct view *view, const char *data, struct diff_state *state)
 		return true;
 
 	} else if (type == LINE_PP_MERGE) {
+		/* we will check again when highlighting diff chunks but we
```

Comment:
Thanks for spotting and fixing this issue. What about removing this part as you initially proposed and changing the test `if (state->combined_diff && !state->after_diff && data[0] == ' ' && data[1] != ' ')` to `if (!state->after_diff && data[0] == ' ' && data[1] != ' ')` as the check on after_diff should be sufficient here ? Maybe it would be wise to add a comment stating that this test is necessary because merge commits lack the LINE_DIFF_START.

Reply:
I agree, that's better.

Reply:
Thanks for the review!